Frameworks/Foundation/NSProgress.mm

```diff
@@ -252,10 +278,22 @@ - (int64_t)completedUnitCount {
 /**
  @Status Interoperable
 */
-- (double)fractionCompleted {
-    // Override synthesized getter so that this can be calculated dynamically
+- (void)setTotalUnitCount:(int64_t)inUnitCount {
+    @synchronized(self) { // Property is atomic
+        double ratio = (double)inUnitCount / _totalUnitCount;
+        _totalUnitCount = inUnitCount;
+        [self _updateCompletedUnitsBy:0
+                  fractionCompletedBy:(_fractionCompleted / ratio) - _fractionCompleted
```

Comment:
>ratio [](start = 60, length = 5)

nit: instead of two divisions, you can inline ratio here, resulting in one division.